src/common/util.c

```diff
@@ -1307,6 +1307,49 @@ safe_strcpy (char *dest, const char *src, int bytes_left)
 	}
 }
 
+const char *
+safe_timestr(time_t ts)
```

Comment:
Would this be more portable and simple?:

```c
GDateTime *datetime = g_date_time_new_from_unix_utc (ts);
GDateTime *localtime = g_date_time_to_local (datetime);
char *timestr = g_date_time_format (datetime, "...");
g_date_time_unref (datetime);
g_date_time_unref (localtime);
return timestr;
```

It results in allocations but thats fine.

Reply:
Docs: https://developer-old.gnome.org/glib/stable/glib-GDateTime.html